File: client/src/models/GameModel.js
import BoardModel from "./BoardModel";
import CastlesModel from "./CastlesModel";

export default class GameModel {
    /** @type {BoardModel} */
    board;
    /** @type {"WHITE" | "BLACK"} */
    turn;
    /** @type {CastlesModel} */
    castles;

    /** 
     * @param {GameModel} data 
    */
    constructor(data) {
        this.board = new BoardModel(data.board);
        this.turn = data.turn;
        this.castles = new CastlesModel(data.castles);
    }
}

File: server/src/main/java/com/victovito/chessapp/models/GameModel.java
package com.victovito.chessapp.models;

import com.victovito.chessapp.enums.Color;
import com.victovito.chessapp.objects.Game;

public class GameModel {
    BoardModel board;
    Color turn;
    CastlesModel castles;
    
    public GameModel() {}

    public GameModel(Game game) {
        board = new BoardModel(game.getBoard());
        turn = game.getTurn();
        castles = new CastlesModel(game.getCastles());
    }

    public BoardModel getBoard() {
        return board;
    }

    public Color getTurn() {
        return turn;
    }

    public CastlesModel getCastles() {
        return castles;
    }
    
}
